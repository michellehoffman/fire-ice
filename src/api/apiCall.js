export const getHouses = async() => {
  try {
    const response = await fetch('http://localhost:3001/api/v1/houses');
    const results = await response.json();
    debugger;

    return results;
  } catch(error) {
    throw Error
  }
}