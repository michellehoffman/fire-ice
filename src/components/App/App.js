import React, { Component } from 'react';
import { func, arrayOf, object } from 'prop-types';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';
import { addHouses } from '../../actions';
import { getHouses } from '../../api/apiCall';
import gif from '../../assets/wolf.gif';
import { Card } from '../Card/Card';

export class App extends Component {
  constructor(props) {
    super(props)

    this.state = {
      loaded: false
    }
  }

  async componentDidMount() {
    const houses = await getHouses();
    this.props.addHouses(houses);
    if(this.props.houses) {
      this.setState({ loaded: true })
    }
  }

  renderCards = () => {
    return this.props.houses.map( (house, index) => (
      <Card key = { index + house.name } { ...house } />)
    )
  }

  render() {
    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westeros</h2>
        </div>
        <div className='Display-info'>
          { !this.state.loaded &&
            <img src={ gif } alt='wolf gif' />
          }
          {
            this.state.loaded &&
              this.renderCards()
          }
        </div>
      </div>
    );
  }
}

App.propTypes = {
  addHouses: func.isRequired,
  houses: arrayOf(object).isRequired
};

export const mapStateToProps = ({ houses }) => ({ houses });

export const mapDispatchToProps = dispatch => ({ 
  addHouses: houses => dispatch(addHouses(houses))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);