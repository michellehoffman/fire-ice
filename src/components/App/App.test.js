import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { getHouses, getSwornMembers } from '../../api/apiCall';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    const mockHouses = [{
      name: '',
      founded: '',
      seats: [''],
      titles: [''],
      coatOfArms: '',
      ancestralWeapons: [''],
      words: '',
      swornMembers: [{}]
    }]

    wrapper = shallow(<App addHouses={ jest.fn() }
                           houses={ mockHouses } />, { disableLifecycleMethods: true })
  })

  it('should match snapshot when houses are NOT loaded', () => {
    wrapper.setState({ loaded: false });

    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot when houses are loaded', () => {
    wrapper.setState({ loaded: true })

    expect(wrapper).toMatchSnapshot();
  });

  it('should map state correctly', () => {
    const mockHouses = [{
      name: '',
      founded: '',
      seats: [''],
      titles: [''],
      coatOfArms: '',
      ancestralWeapons: [''],
      words: '',
      swornMembers: [{}]
    }]
    const mockStore = {
      houses: mockHouses
    }
    const expected = mockStore

    expect(mapStateToProps(mockStore)).toEqual(expected);
  });

  it('should call dispatch when an action is called', () => {
    const mockDispatch = jest.fn();

    mapDispatchToProps(mockDispatch).addHouses();

    expect(mockDispatch).toHaveBeenCalled();
  });
  
});
