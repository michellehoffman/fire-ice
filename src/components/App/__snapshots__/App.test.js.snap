// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot when houses are NOT loaded 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addHouses={[Function]}
    houses={
        Array [
            Object {
              "name": "House Corbray of Heart's Home",
            },
            Object {
              "name": "House Dayne of Starfall",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h2>
                    Welcome to Westeros
          </h2>
</div>,
        <div
          className="Display-info"
>
          <img
                    alt="wolf gif"
                    src="wolf.gif"
          />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h2>
              Welcome to Westeros
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Westeros",
            },
            "ref": null,
            "rendered": "Welcome to Westeros",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="wolf gif"
              src="wolf.gif"
/>,
            false,
          ],
          "className": "Display-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "wolf gif",
              "src": "wolf.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h2>
                        Welcome to Westeros
            </h2>
</div>,
          <div
            className="Display-info"
>
            <img
                        alt="wolf gif"
                        src="wolf.gif"
            />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h2>
                Welcome to Westeros
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Westeros",
              },
              "ref": null,
              "rendered": "Welcome to Westeros",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="wolf gif"
                src="wolf.gif"
/>,
              false,
            ],
            "className": "Display-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "wolf gif",
                "src": "wolf.gif",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`App should match snapshot when houses are loaded 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addHouses={[Function]}
    houses={
        Array [
            Object {
              "name": "House Corbray of Heart's Home",
            },
            Object {
              "name": "House Dayne of Starfall",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h2>
                    Welcome to Westeros
          </h2>
</div>,
        <div
          className="Display-info"
>
          <Card
                    name="House Corbray of Heart's Home"
          />
          <Card
                    name="House Dayne of Starfall"
          />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h2>
              Welcome to Westeros
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Westeros",
            },
            "ref": null,
            "rendered": "Welcome to Westeros",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            Array [
              <Card
                name="House Corbray of Heart's Home"
/>,
              <Card
                name="House Dayne of Starfall"
/>,
            ],
          ],
          "className": "Display-info",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": "0House Corbray of Heart's Home",
            "nodeType": "class",
            "props": Object {
              "name": "House Corbray of Heart's Home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1House Dayne of Starfall",
            "nodeType": "class",
            "props": Object {
              "name": "House Dayne of Starfall",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h2>
                        Welcome to Westeros
            </h2>
</div>,
          <div
            className="Display-info"
>
            <Card
                        name="House Corbray of Heart's Home"
            />
            <Card
                        name="House Dayne of Starfall"
            />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h2>
                Welcome to Westeros
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Westeros",
              },
              "ref": null,
              "rendered": "Welcome to Westeros",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              Array [
                <Card
                  name="House Corbray of Heart's Home"
/>,
                <Card
                  name="House Dayne of Starfall"
/>,
              ],
            ],
            "className": "Display-info",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": "0House Corbray of Heart's Home",
              "nodeType": "class",
              "props": Object {
                "name": "House Corbray of Heart's Home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1House Dayne of Starfall",
              "nodeType": "class",
              "props": Object {
                "name": "House Dayne of Starfall",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
